packages/rx_bloc_cli/mason_templates/bricks/lib_auth_apple/__brick__/lib/lib_auth_apple/di/apple_login_button_with_dependencies.dart

```diff
@@ -0,0 +1,78 @@
+{{> licence.dart }}
+
+import 'package:flutter/widgets.dart';
+import 'package:flutter_rx_bloc/flutter_rx_bloc.dart';
+import 'package:provider/provider.dart';
+
+import '../../base/common_ui_components/app_error_modal_widget.dart';
+import '../../base/data_sources/remote/http_clients/api_http_client.dart';
+import '../blocs/apple_login_bloc.dart';
+import '../data_sources/apple_auth_data_source.dart';
+import '../data_sources/apple_credential_data_source.dart';
+import '../repositories/apple_auth_repository.dart';
+import '../services/apple_social_login_service.dart';
+import '../ui_components/apple_login_button.dart';
+
+/// [AppleLoginButtonWithDependencies] provides out of the box Log in with Apple
+/// functionality along with default view of the button. If you want to customize
+/// the way button looks use [builder].
+///
+/// If an error occur a modal sheet with message will be shown. For custom error
+/// handling provide [onError] callback.
+class AppleLoginButtonWithDependencies extends StatelessWidget {
+  const AppleLoginButtonWithDependencies({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) => MultiProvider(
+        providers: [
+          ..._dataSources,
+          ..._repositories,
+          ..._services,
+          ..._blocs,
+        ],
+        child: Column(
+          children: [
+            AppErrorModalWidget<AppleLoginBlocType>(
+              errorState: (bloc) => bloc.states.errors,
+            ),
+            RxBlocBuilder<AppleLoginBlocType, bool>(
+              state: (bloc) => bloc.states.isLoading,
+              builder: (context, snapshot, bloc) => const AppleLoginButton(),
```

Comment:
Why we use the `RxBlocBuilder` if we do not check for the state ?

Reply:
A new custom button with loading state will be created. But until then I will remove this.

Reply:
Removed the builder.